thriftbp/server_test.go

```diff
@@ -0,0 +1,62 @@
+package thriftbp_test
```

Comment:
OK, so in terms of unblocking:

* If we don't need an API, or at least yet, I would prefer to not have one so that we don't complicate any future programmatic migrations to a new baseplate version.  The question of whether we need this API seems to come back to the question of whether the current thriftbp can work for your service.
* If we need an API, I think the `With`-style functional options are a good choice generally, but the details of usage matter, and I haven't seen the example PR that makes use of this so I am purely extrapolating.
* Both of these examples show using a _lot_ of options.  When a lot of callers are using a lot of options, the functional options actually aren't that great of a choice in many cases -- they're optimal for callers who need 0 or 1 option, with the rare exception of those who need more.  The 
* Most examples should be targeted at actual usage patterns.  If this is what it will actually look like for callers who need to customize one thing -- i.e. having to spell out a lot of options manually -- then I think it's going the wrong way.  If this is just to make use of all of the available options, then I wouldn't worry about that -- the options are right there in the godoc.  Make these examples focus on what actual users will need, if that makes sense.
* At the end of the day, if we don't want people to use these APIs except out of dire need, we need to discourage it.  Maybe We need to add "Dangerous" or "Unsupported" or something to the ones that we don't want people to use.